perlfunc: document that sort is stable, remove 'use sort' references

mauke opened the pull request.

mauke committed:
perlfunc: document that sort is stable, remove 'use sort' references

Grinnz commented:
It's definitely worth documenting that this is how it currently works, but do we want to provide a guarantee? I suppose at this point it would be just as difficult to change that either way.

jkeenan commented:
> It's definitely worth documenting that this is how it currently works, but do we want to provide a guarantee? I suppose at this point it would be just as difficult to change that either way.

The `DESCRIPTION` section of `perldoc sort` is quite emphatic that we're not changing how `sort` works:
```
    The default sort has been stable since v5.8.0, and given this consistent
    behaviour for almost two decades, everyone has come to assume stability.

    Stability will remain the default - hence there is no need for a pragma
    for code to opt into stability "just in case" this changes - it won't.

    We do not foresee going back to offering multiple implementations of
    general purpose sorting - hence there is no future need to offer a
    pragma to choose between them.
...
    We are not averse to changing the sort algorithm, but we don't see the
    benefit in offering the choice of two general purpose implementations.
```

saeedaziimi commented:
****

mauke committed:
sort: fix display of verbatim blocks by using spaces, not tabs

The `use sort 'defaults';` line used only one tab before the comment;
the other lines had two. This renders inconsistently depending on how
much initial space is used in the output (and where the tab stops are).

For example, <https://perldoc.perl.org/sort#SYNOPSIS> looked like

    use sort 'stable';              # guarantee stability
    use sort 'defaults';    # revert to default behavior
    no  sort 'stable';              # stability not important

Using spaces ensures the "#" signs are aligned everywhere.
